app/controllers/users_controller.rb

```diff
@@ -1,6 +1,28 @@
 class UsersController < ApplicationController
   def show
-    @wikis = current_user.wikis.all
-    @wiki = current_user.wikis.new
+    @wikis = private_check.paginate(page: params[:page], per_page: 10)
+    @wiki = Wiki.new
   end
-end
\ No newline at end of file
+
+  def update
+    if current_user.update_attributes(user_params)
+      flash[:notice] = "User information updated."
+      redirect_to current_user
+    else
+      flash[:error] = "There was an error updating user information."
+      redirect_to current_user
+    end
+  end
+
+  def private_check
```

Comment:
You should probably do this on downgrade as opposed to every time a standard user hits this page.